src/builder.ts

```diff
@@ -208,24 +221,16 @@ export default class Builder {
 		handler: ErrorHandler,
 		...args: any[]
 	): Bluebird<any> {
-		if (hook in hooks) {
-			try {
+		return Bluebird.try(() => {
+			const fn = hooks[hook];
+			if (_.isFunction(fn)) {
 				// Spread the arguments onto the callback function
-				const fn = hooks[hook];
-				if (_.isFunction(fn)) {
-					const val = fn.apply(null, args);
-					// If we can add a catch handler
-					if (val != null && _.isFunction(val.catch) && _.isFunction(handler)) {
-						val.catch(handler);
```

Comment:
`val.catch(handler)` here meant that the promised returned by `callHook` was resolved instead of rejected, although `handler` would be called with an error. I've simplified the code with the use of `Bluebird.try` and `tapCatch`, so now `handler` is called if and only if the promise is rejected.